buildstockbatch/cloud/docker_base.py

```diff
@@ -183,3 +188,149 @@ def prep_batches(self, tmppath):
         os.makedirs(tmppath / "results" / "simulation_output")
 
         return (job_count, unique_epws)
+
+    @classmethod
+    def get_epws_to_download(cls, sim_dir, jobs_d):
+        """
+        Gets the list of filenames for the weather data required for a single batch of simulations.
+
+        :param sim_dir: Path to the directory where job files are stored
+        :param jobs_d: Contents of a single job JSON file; contains the list of buildings to simulate in this job.
+
+        :returns: Set of epw filenames needed for this batch of simulations.
+        """
+        # Make a lookup of which parameter points to the weather file from options_lookup.tsv
+        with open(sim_dir / "lib" / "resources" / "options_lookup.tsv", "r", encoding="utf-8") as f:
+            tsv_reader = csv.reader(f, delimiter="\t")
+            next(tsv_reader)  # skip headers
+            param_name = None
+            epws_by_option = {}
+            for row in tsv_reader:
+                row_has_epw = [x.endswith(".epw") for x in row[2:]]
+                if sum(row_has_epw):
+                    if row[0] != param_name and param_name is not None:
+                        raise RuntimeError(
+                            "The epw files are specified in options_lookup.tsv under more than one parameter type: "
+                            f"{param_name}, {row[0]}"
+                        )
+                    epw_filename = row[row_has_epw.index(True) + 2].split("=")[1]
+                    param_name = row[0]
+                    option_name = row[1]
+                    epws_by_option[option_name] = epw_filename
+
+        # Look through the buildstock.csv to find the appropriate location and epw
+        epws_to_download = set()
+        building_ids = [x[0] for x in jobs_d["batch"]]
+        with open(
+            sim_dir / "lib" / "housing_characteristics" / "buildstock.csv",
+            "r",
+            encoding="utf-8",
+        ) as f:
+            csv_reader = csv.DictReader(f)
+            for row in csv_reader:
+                if int(row["Building"]) in building_ids:
+                    epws_to_download.add(epws_by_option[row[param_name]])
+
+        return epws_to_download
+
+    @classmethod
+    def run_simulations(cls, cfg, job_id, jobs_d, sim_dir, fs, bucket, prefix):
```

Comment:
This function doesn't need to care about cloud storage schema, so consider consolidating `bucket` & `prefix`; e.g., just pass in `sim_out_dir`.

Reply:
Good call - updated.